src/ppx/ppx_eliom_utils.ml

```diff
@@ -461,8 +469,277 @@ end
+          Exp.fun_ (Asttypes.Optional s) None pattern (expr_mapper expr')
+      | _ ->
+          let rpc_name =
+            Exp.constant (Pconst_string (rpc_name fun_name_pattern, None))
+          in
+          let fun_name = function_name fun_name_pattern in
+          let aux_apply =
+            apply_function_expr ~loc "aux" [(Asttypes.Nolabel, _expr_of_args)]
+          in
+          apply_rpc ~loc rpc_name fun_name args_list _pat_of_args _typ_of_args aux_apply
+    in
+    expr_mapper expr
+
+  let server_side rpc_type stri =
+    let loc = stri.pstr_loc in
+    match stri with
+    | [%stri
+        let [%p? {ppat_desc= Ppat_var ident} as pattern] =
+          [%e? [%expr fun [%p? _] -> [%e? expr']]]] ->
+        let args_list = args_parser expr' [] in
+        let _args =
+          List.map (fun (pattern, _, _) -> pattern) args_list |> pat_args
+        in
+        let _apply =
+          match rpc_type with
+          | `Connected_rpc_o ->
+              ( Asttypes.Nolabel
+              , [%expr Os_current_user.Opt.get_current_userid ()] )
+              :: List.map
+                   (fun (pattern, _, label) -> (label, pat_to_exp pattern))
+                   args_list
+          | _ ->
+              (Asttypes.Nolabel, [%expr Os_current_user.get_current_userid ()])
+              :: List.map
+                   (fun (pattern, _, label) -> (label, pat_to_exp pattern))
+                   args_list
+        in
+        ([ [%stri
+             let [%p pattern] =
+               [%expr
+                 fun [%p _args] ->
+                   [%e apply_function_expr ~loc ident.txt _apply]]]
+        ] [@metaloc loc])
+    | _ -> (
+        ([stri] [@metaloc loc]) )
+
+  (*let rpc_server rpc_type stri =
+  let loc = stri.pstr_loc in
+  match stri with
+  | [%stri let [%p? {ppat_desc= Ppat_var ident} as pattern] = [%e? expr]] -> (
+    match expr with
+    | [%expr fun [%p? _] -> [%e? expr']] ->
+        let args_list = args_parser expr' [] in
+        let _args =
+          List.map (fun (pattern, _, _) -> pattern) args_list |> format_pattern
+        in
+        let list = match rpc_type with 
+        |`Connected_rpc_o ->
+          (Asttypes.Nolabel, [%expr Os_current_user.Opt.get_current_userid ()])
+          :: List.map
+               (fun (pattern, _, label) -> (label, pat_to_exp pattern))
+               args_list
+        |_ ->(Asttypes.Nolabel, [%expr Os_current_user.get_current_userid ()])
+          :: List.map
+               (fun (pattern, _, label) -> (label, pat_to_exp pattern))
+               args_list
+        in
+        let apply =
+          Exp.apply
+            (Exp.ident (Location.mkloc (Longident.parse ident.txt) loc))
+            list
+        in
+        ([[%stri let [%p pattern] = fun [%p _args] -> [%e apply]]][@metaloc loc])
+    | _ ->
+        print_error ~loc Fatal_error )
+  | _ ->
+      print_error ~loc Fatal_error*)
+
+  let client_side rpc_type stri =
```

Comment:
I would advise you to prefix all functions building ast parts by "generate_".
As functions are actions to be done, it is nice to use a action verb to express this. This is not an absolute rule and there can be many exceptions. Here when I read `client_side`, before diving into your code I tell myself "client side what ?".